src/dataflow/graph/graph.ts

```diff
@@ -177,6 +179,33 @@ export class DataflowGraph {
 		return this
 	}
 
+
+	/**
+	 * Adds a vertex for variable use. Intended for creating dataflow graphs as part of function tests.
+	 * @param id - AST node id
+	 * @param name - Variable name
+	 * @param when - always or maybe; defaults to always
+	 * @param environment - Environment the use occurs in
+	 * @param asRoot - boolean; defaults to true.
+	 * @param scope - string
+	 */
+	public uses(id: string, name: string, when: DataflowGraphEdgeAttribute = 'always', environment?: REnvironmentInformation, asRoot: boolean = true) {
+		return this.addVertex({ tag: 'use', id, name, when, environment }, asRoot)
+	}
+
+	/**
+	 * Adds a vertex for a variable definition.
+	 * @param id - AST node ID
+	 * @param name - Variable name
+	 * @param scope - Scope (global/local/custom), defaults to local.
+	 * @param when - always or maybe; defaults to maybe.
+	 * @param environment - Environment the variable is defined in.
+	 * @param asRoot - boolean; defaults to true.
+	 */
+	public definesVariable(id: string, name: string, scope: string = LocalScope, when: DataflowGraphEdgeAttribute = 'always', environment?: REnvironmentInformation, asRoot: boolean = true) {
```

Comment:
Is there a reason you do not use the `NodeId` type alias for the id? I consider changing the type.

Reply:
Besides I am not too sure about the name `definesVariable` (which is long and focuses on the variable which is missing from a function like `uses`). 
How do you like `define(variable|function, ...)`, `use`, ...?